cmd/internal/opentelemetry/wrapperMux.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentelemetry
+
+import (
+	"flag"
+	"fmt"
+	"github.com/google/cadvisor/version"
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/correlation"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/propagation"
+	apiTrace "go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/plugin/othttp"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"google.golang.org/grpc/credentials"
+	"k8s.io/klog/v2"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+var (
+	traceSampler      = flag.String("trace_sampler", "always", "choose a sampler 'never', 'always' or 'probability'")
+	traceSamplerParam = flag.Float64("trace_sampler_param", 0, "used for the 'probability' sampler")
+	traceServiceName  = flag.String("trace_name", "cadvisor", "use localEndpoint service name")
+	traceTags         tracingTags
+)
+
+func init() {
+	flag.Var(&traceTags, "trace_tags", "additional tags for tracing (eg. foo=bar)")
+}
+
+func InitTrace(traceEndpoint, collectorCert string) *otlp.Exporter {
+	exporterOptions := []otlp.ExporterOption{otlp.WithAddress(traceEndpoint)}
+
+	if collectorCert != "" {
+		if creds, err := credentials.NewClientTLSFromFile(collectorCert, ""); err != nil {
+			klog.Fatal(err)
+		} else {
+			exporterOptions = append(exporterOptions, otlp.WithTLSCredentials(creds))
+		}
+	} else {
+		exporterOptions = append(exporterOptions, otlp.WithInsecure())
+	}
+
+	if exporter, err := otlp.NewExporter(exporterOptions...); err != nil {
+		klog.Fatal(err)
+	} else {
+		sampler := getTraceSampler(*traceSampler, *traceSamplerParam)
+		providerOpts := []trace.ProviderOption{
+			trace.WithBatcher(exporter),
+			trace.WithConfig(trace.Config{DefaultSampler: sampler}),
+			trace.WithResourceAttributes(getTraceTags(traceTags)...),
+		}
+
+		if traceProvider, err := trace.NewProvider(providerOpts...); err != nil {
+			klog.Fatal(err)
+		} else {
+			global.SetTraceProvider(traceProvider)
+
+			apiB3, traceCtx, corrCtx := apiTrace.B3{SingleHeader: false}, apiTrace.DefaultHTTPPropagator(), correlation.DefaultHTTPPropagator()
+			options := []propagation.Option{propagation.WithInjectors(apiB3, traceCtx, corrCtx),
```

Comment:
i'm surprised we need to specify the propagation format...  I don't see this done in any examples.  Are you sure this is required?

Reply:
The default implementation dosn't use the B3 headers.

https://github.com/open-telemetry/opentelemetry-go/blob/master/api/global/internal/state.go#L124

I'm not happy with that implementation, I have to find a independend solution or more flags 
:(

Reply:
I added a own HTTPExtracor, which checks all available Extractors.

It is not possible to register all HTTPExtractor, because it override a prepared Context, if the next failed.

Reply:
Dumb question, but is there a difference between the W3C and B3 format?

Reply:
Yes, the [W3C Trace Context Recommendation](https://www.w3.org/TR/trace-context/#traceparent-header) specifies a `traceparent` header that includes the trace ID, parent span ID, and trace flags.  The [B3 format](https://github.com/openzipkin/b3-propagation#http-encodings) specifies both multiple-header and single-header encodings of the same information with provision for additional information.  The single-header B3 encoding is nearly identical to the W3C `traceparent` encoding, but for the `version` field and `flags` vs. `sampled`.